flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BarrierBuffer.java

```diff
@@ -254,8 +357,20 @@ public void cleanup() throws IOException {
 		for (BufferSpiller.SpilledBufferOrEventSequence seq : queuedBuffered) {
 			seq.cleanup();
 		}
+		queuedBuffered.clear();
 	}
-	
+
+	private void beginNewAlignment(long checkpointId, int channelIndex) throws IOException {
+		currentCheckpointId = checkpointId;
+		onBarrier(channelIndex);
+
+		startOfAlignmentTimestamp = System.nanoTime();
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Starting stream alignment for checkpoint " + checkpointId);
```

Comment:
When guarded this here is actually the more efficient pattern. So, i'd lean towards ignoring/deactivating that warning.